### PR TITLE
Cassandra 12988

### PR DESCRIPTION
**Pull request by jmckenzie-dev**

### CODE REVIEW

**Review by blerer — COMMENTED** [review]
My concern with the current patch is that we keep the overiding values in a separate location (`AuthProperties`) instead of updating them in `DatabaseDescriptor` as it is usually done. If somebody use the `Settings` system view to look at the current settings he will see the default setting and not the updated ones. That can be confusing in my opinion.

**Comment by jmckenzie-dev** [comment]
> My concern with the current patch is that we keep the overiding values in a separate location (`AuthProperties`) instead of updating them in `DatabaseDescriptor` as it is usually done. If somebody use the `Settings` system view to look at the current settings he will see the default setting and not the updated ones. That can be confusing in my opinion.

Totally agree; this was an artifact of me taking an old approach and pushing and pulling it until I had a good grasp of where the static state was initialized during clientInitialization.

I've pushed up another commit that makes Config the canonical owner of auth read and write CL state and keeps the String <-> ConsistencyLevel conversion barrier between DBD and Config, removing all state tracking from AuthProperties.

Let me know what you think.

File: src/java/org/apache/cassandra/auth/CassandraRoleManager.java [hunk 1]
```diff
@@ -508,6 +507,23 @@ private static String escape(String name)
         return StringUtils.replace(name, "'", "''");
     }
 
+    /** Allows selective overriding of the consistency level for specific roles. */
+    protected static ConsistencyLevel consistencyForRoleWrite(String role)
+    {
+        if (role.equals(DEFAULT_SUPERUSER_NAME))
+            return DEFAULT_SUPERUSER_CONSISTENCY_LEVEL;
+        else
+            return CassandraAuthorizer.authWriteConsistencyLevel();
```
Comment (blerer) [inline src/java/org/apache/cassandra/auth/CassandraRoleManager.java]: nit: I feel that a ternary operator might be more readable. The `else` between 2 return statements is confusing me.
Same thing in the next method.

File: src/java/org/apache/cassandra/config/Config.java [hunk 2]
```diff
@@ -515,6 +515,11 @@
      */
     public volatile int validation_preview_purge_head_start_in_sec = 60 * 60;
 
+    // Using String instead of ConsistencyLevel here to keep static initialization from cascading and starting
+    // threads during tool usage mode. See CASSANDRA-12988 and DatabaseDescriptorRefTest for details
+    public String auth_read_consistency_level = "LOCAL_QUORUM";
+    public String auth_write_consistency_level = "EACH_QUORUM";
```
Comment (blerer) [inline src/java/org/apache/cassandra/config/Config.java]: As those variables can be accessed and modify by multiple threads they should be `volatile`.

**Pull request closed without merging**